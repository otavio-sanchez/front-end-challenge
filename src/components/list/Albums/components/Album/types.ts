import { Album } from '../../types';

export interface Props extends Album {
    height?: number;
}
