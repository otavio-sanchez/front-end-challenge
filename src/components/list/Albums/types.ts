export interface Props {
    data: Album[];
    title: string;
    height?: number;
}

export interface Album {
    image: string;
    redirect: string;
    id: string;
    name: string;
    artist: string;
}
