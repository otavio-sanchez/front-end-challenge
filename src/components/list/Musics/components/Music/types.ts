export interface Props {
    id: string;
    name: string;
    time: number;
    trackNumber: number;
    onClick: Function;
}
