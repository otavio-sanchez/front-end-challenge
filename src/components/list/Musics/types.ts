export interface Props {
    data: Music[];
    onClickItem: Function;
}

export interface Music {
    id: string;
    name: string;
    time?: number;
    preview_url: string;
    track_number: number;
    duration_ms?: number;
}
