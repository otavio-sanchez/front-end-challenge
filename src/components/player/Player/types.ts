export interface Props {
    images: Image[];
    id: string;
    name: string;
    artists: Artist[];
    onClick: Function;
    preview: string;
}

export interface Artist {
    id: string;
    name: string;
}

export interface Image {
    height: number;
    url: string;
    width: number;
}
