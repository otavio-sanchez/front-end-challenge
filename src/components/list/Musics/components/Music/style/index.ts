import styled, { keyframes } from 'styled-components';

const opacityAnimation = keyframes`
  from {
    opacity: 0;
  }

  to {
    opacity: 1;
  }
`;

const Content = styled.div`
    animation: ${opacityAnimation} 1s linear;
    width: 100%;
    cursor: pointer;
`;

const Image = styled.div<{ src: string; height?: number }>`
    width: 100%;
    height: ${({ height }): number => (height ? height : 250)}px;
    background-image: url('${({ src }): string => src}');
    background-size: cover;    
    transition: opacity ease-in-out 0.5s;
    background-position: center;
    opacity: 1;
    &:hover {
        transition: opacity ease-in-out 0.5s;
        opacity: 0.5;
    }
`;

const Infos = styled.div`
    text-align: start;
    padding: 20px 0;
    width: 100%;
    box-sizing: border-box;
`;

const TrackNumber = styled.span`
    padding-right: 20px;
`;

const Time = styled.span`
    padding-left: 20px;
    float: right;
`;

export { Content, Image, Infos, TrackNumber, Time };
