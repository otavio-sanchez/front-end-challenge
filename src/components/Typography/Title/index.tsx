import * as React from 'react';
import { TitleLarge } from './style';
import { Props } from './types';

const Title = (props: Props) => <TitleLarge>{props.text}</TitleLarge>;

export default Title;
