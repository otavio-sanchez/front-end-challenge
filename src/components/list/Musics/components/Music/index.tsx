import * as React from 'react';
import { Text } from '../../../../typography';
import { Content, Infos, TrackNumber, Time } from './style';
import { Props } from './types';
import { millisToMinutesAndSeconds } from '../../../../../utils/convertHours';

const Music = ({ name, time, trackNumber, onClick }: Props): JSX.Element => (
    <Content onClick={(): Function => onClick()}>
        <Infos>
            <TrackNumber>
                <Text type="regular">
                    <>{trackNumber}.</>
                </Text>
            </TrackNumber>
            <Text type="default">{name}</Text>
            <Time>
                <Text type="regular">{millisToMinutesAndSeconds(time)}</Text>
            </Time>
        </Infos>
    </Content>
);

export default Music;
