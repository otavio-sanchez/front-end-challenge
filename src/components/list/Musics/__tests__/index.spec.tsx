import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Musics } from '../..';
import { Props } from '../types';
import musics from '../../../../constants/mocks/musics';
import Music from '../components/Music';
import { Text } from '../../../Typography';

describe('Musics', () => {
    const props: Props = {
        data: musics,
        onClickItem: () => {},
    };

    it('renders without crashing given the required props', () => {
        const wrapper = shallow(<Musics {...props} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render three Album component', () => {
        const wrapper = shallow(<Musics {...props} />);
        expect(wrapper.find(Music)).toHaveLength(2);
    });

    it('should render feedback when there is not data', () => {
        const wrapper = shallow(<Musics {...props} />);
        wrapper.setProps({ data: [] });
        console.log(wrapper.find(Text).get(0).props.children);
        expect(wrapper.find(Text).get(0).props.children).toContain('Nada encontrado.');
    });
});
