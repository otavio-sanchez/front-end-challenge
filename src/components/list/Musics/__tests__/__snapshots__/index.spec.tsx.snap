// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Albums renders without crashing given the required props 1`] = `
<Fragment>
  <Container>
    <Row
      columnsDesktop={1}
      columnsMobile={1}
      columnsTablet={1}
      spacingColumn={30}
      spacingRow={30}
    >
      <Col>
        <Text
          type="title"
        >
           
          Met√°lica
           
        </Text>
      </Col>
    </Row>
    <Row
      columnsDesktop={5}
      columnsMobile={1}
      columnsTablet={2}
      spacingColumn={30}
      spacingRow={30}
    >
      <Col
        key="0"
      >
        <Album
          artists={
            Array [
              Object {
                "id": "0",
                "name": "artist foo",
              },
            ]
          }
          height={360}
          id="0"
          images={
            Array [
              Object {
                "height": 640,
                "url": "https://i.scdn.co/image/ab67616d0000b273af07dc851962508661bbcfce",
                "width": 640,
              },
              Object {
                "height": 300,
                "url": "https://i.scdn.co/image/ab67616d00004851af07dc851962508661bbcfce",
                "width": 300,
              },
              Object {
                "height": 64,
                "url": "https://i.scdn.co/image/ab67616d0000b273af07dc851962508661bbcfce",
                "width": 64,
              },
            ]
          }
          name="Album foo"
          onClick={[Function]}
        />
      </Col>
      <Col
        key="1"
      >
        <Album
          artists={
            Array [
              Object {
                "id": "1",
                "name": "artist bar",
              },
            ]
          }
          height={360}
          id="1"
          images={
            Array [
              Object {
                "height": 640,
                "url": "https://i.scdn.co/image/ab67616d0000b273af07dc851962508661bbcfce",
                "width": 640,
              },
              Object {
                "height": 300,
                "url": "https://i.scdn.co/image/ab67616d00004851af07dc851962508661bbcfce",
                "width": 300,
              },
              Object {
                "height": 64,
                "url": "https://i.scdn.co/image/ab67616d0000b273af07dc851962508661bbcfce",
                "width": 64,
              },
            ]
          }
          name="Album bar"
          onClick={[Function]}
        />
      </Col>
      <Col
        key="2"
      >
        <Album
          artists={
            Array [
              Object {
                "id": "2",
                "name": "artist baz",
              },
            ]
          }
          height={360}
          id="2"
          images={
            Array [
              Object {
                "height": 640,
                "url": "https://i.scdn.co/image/ab67616d0000b273af07dc851962508661bbcfce",
                "width": 640,
              },
              Object {
                "height": 300,
                "url": "https://i.scdn.co/image/ab67616d00004851af07dc851962508661bbcfce",
                "width": 300,
              },
              Object {
                "height": 64,
                "url": "https://i.scdn.co/image/ab67616d0000b273af07dc851962508661bbcfce",
                "width": 64,
              },
            ]
          }
          name="Album baz"
          onClick={[Function]}
        />
      </Col>
    </Row>
  </Container>
</Fragment>
`;

exports[`Musics renders without crashing given the required props 1`] = `
<Container>
  <Row>
    <Music
      id="5BIMPccDwShpXq784RJlJp"
      key="5BIMPccDwShpXq784RJlJp"
      name="Enter Sandman"
      onClick={[Function]}
      time={331573}
      trackNumber={1}
    />
    <Music
      id="1Ug98CRiQYu7wiSym4uLY4"
      key="1Ug98CRiQYu7wiSym4uLY4"
      name="Sad But True"
      onClick={[Function]}
      time={324560}
      trackNumber={2}
    />
  </Row>
</Container>
`;
